scio-examples/src/test/scala/com/spotify/scio/examples/extra/HttpAsyncLookupExampleTest.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.examples.extra
+
+import java.io.IOException
+import java.net.InetSocketAddress
+
+import com.google.common.util.concurrent.MoreExecutors
+import com.spotify.scio.io._
+import com.spotify.scio.testing._
+import com.sun.net.httpserver.{HttpExchange, HttpHandler, HttpServer}
+import org.scalatest.BeforeAndAfterAll
+
+class HttpAsyncLookupExampleTest extends PipelineSpec with BeforeAndAfterAll {
+
+  override protected def beforeAll(): Unit = SimpleHttpServer.start
+
+  override protected def afterAll(): Unit = SimpleHttpServer.stop
+
+  "HttpAsyncLookupExample" should "work for a successful request" in {
+    val expected = (1 to 5).map { i =>
+      s"$i\t${i.toDouble}\ttest"
+    }
+
+    JobTest[com.spotify.scio.examples.extra.HttpAsyncLookupExample.type]
+      .args("--output=out.text", "--url=http://localhost:9000/success")
+      .output(TextIO("out.text"))(_ should containInAnyOrder(expected))
+      .run()
+  }
+
+  it should "work with server error" in {
+    val expected = (1 to 5).map { i =>
+      s"$i\t${i.toDouble}\tDefault Value"
+    }
+
+    JobTest[com.spotify.scio.examples.extra.HttpAsyncLookupExample.type]
+      .args("--output=out.text", "--url=http://localhost:9000/server-error")
+      .output(TextIO("out.text"))(_ should containInAnyOrder(expected))
+      .run()
+  }
+
+  it should "work with thrown exception" in {
+    val expected = (1 to 5).map { i =>
+      s"$i\t${i.toDouble}\tFailed"
+    }
+
+    JobTest[com.spotify.scio.examples.extra.HttpAsyncLookupExample.type]
+      .args("--output=out.text", "--url=http://localhost:9000/exception")
+      .output(TextIO("out.text"))(_ should containInAnyOrder(expected))
+      .run()
+  }
+}
+
+object SimpleHttpServer {
+  lazy val server = HttpServer.create(new InetSocketAddress(9000), 0)
```

Comment:
Usually a good idea to randomize ports in tests to avoid collisions.

Reply:
Noted.